official/resnet/resnet_run_loop.py

```diff
@@ -86,21 +97,23 @@ def process_record_dataset(dataset, is_training, batch_size, shuffle_buffer,
   # last batch of an epoch-- do not raise an error when we try to split them
   # over the GPUs. This will likely be handled by Estimator during replication
   # in the future, but for now, we just drop the leftovers here.
-  if multi_gpu:
+  if use_distribution_strategy:
     total_examples = num_epochs * examples_per_epoch
-    dataset = dataset.take(batch_size * (total_examples // batch_size))
+    dataset = dataset.take(
+        per_device_batch_size * (total_examples // per_device_batch_size))
```

Comment:
Is this still necessary? I would imagine not? This was originally a fix for the fact that replicate_model_fn would error out, as noted in the comment (which should also be updated in the unexpected case that this is still relevant). It would be great to remove this, because then we can not pass multi-gpu knowledge this far in, and compute batch size closer to the top of the processing, pass in the desired batch_size here without this func caring whether it's global or per device.